src/sourceMaps/sourceMapUrl.ts

```diff
@@ -0,0 +1,57 @@
+/*---------------------------------------------------------
+ * Copyright (C) Microsoft Corporation. All rights reserved.
+ *--------------------------------------------------------*/
+
+ import { PossiblyCustomerContent, CustomerContent, NonCustomerContent } from '../chrome/logging/gdpr';
+import { logger } from 'vscode-debugadapter';
+import { SourceMapContents } from './sourceMapContents';
+
+export class SourceMapUrl {
+    private readonly _url: PossiblyCustomerContent<string>;
+    public readonly isInlineSourceMap: boolean;
+
+    public constructor(url: string) {
+        this.isInlineSourceMap = url.indexOf('data:application/json') >= 0;
+        this._url = this.isInlineSourceMap ? new CustomerContent(url) : new NonCustomerContent(url);
+    }
+
+    public static maybeUndefined(url: string | undefined): SourceMapUrl | undefined {
```

Comment:
duplicated code as mentioned above

Reply:
I changed the code, it's not duplicated any more